pyflakes/test/test_imports.py

```diff
@@ -564,6 +563,15 @@ def bar():
         fu
         ''', m.UnusedImport, m.UndefinedName)
 
+    def test_shadowedInNestedScope(self):
+        self.flakes('''
+        import fu
+        def bar():
+            import fu
+            fu
+        fu
+        ''', m.ImportShadowedByImport)
```

Comment:
Unused imports is also style, not an error.

I could exlude ast.ImportFrom from this new rule to avoid some scenarios where redefining an imported name is less problematic, such as your example.